index.html

```diff
@@ -61,27 +61,323 @@
+                <span class="card-badge">20% OFF</span>
+              </div>
+              <div class="card-details">
+                <h2 class="detail-heading">Drools for Dogs</h2>
+                <p class="detail-subheading">Adult 1.2 Kg</p>
+              </div>
+              <div class="card-footer">
+                <div class="price-segment">
+                  <span class="item-price">&#x20b9; 250</span>
+                  <span class="item-price dashed">&#x20b9; 350</span>
+                </div>
+                <div class="card-icons">
+                  <i class="bi bi-heart"></i>
+                  <i class="bi bi-cart"></i>
+                </div>
+              </div>
+            </div>
+          </section>
+          <section class="component">
+            <div class="card__container">
+              <div class="card-header">
+                <img
+                  src="/images/pedigree-image.png"
+                  alt="product image"
+                  class="card-image"
+                />
+                <span class="card-badge">20% OFF</span>
+              </div>
+              <div class="card-details">
+                <h2 class="detail-heading">Pedigree for Dogs</h2>
+                <p class="detail-subheading">Adult 1.2 Kg</p>
+              </div>
+              <div class="card-footer">
+                <div class="price-segment">
+                  <span class="item-price">&#x20b9; 250</span>
+                  <span class="item-price dashed">&#x20b9; 350</span>
+                </div>
+                <div class="card-icons">
+                  <i class="bi bi-heart"></i>
+                  <i class="bi bi-cart"></i>
+                </div>
+              </div>
+            </div>
+          </section>
+          <section class="component">
+            <div class="card__container">
+              <div class="card-header">
+                <img src="/images/drools-image.png" alt="product image" class="card-image" />
+                <span class="card-badge">20% OFF</span>
+              </div>
+              <div class="card-details">
+                <h2 class="detail-heading">Drools for Dogs</h2>
+                <p class="detail-subheading">Adult 1.2 Kg</p>
+              </div>
+              <div class="card-footer">
+                <div class="price-segment">
+                  <span class="item-price">&#x20b9; 250</span>
+                  <span class="item-price dashed">&#x20b9; 350</span>
+                </div>
+                <div class="card-icons">
+                  <i class="bi bi-heart"></i>
+                  <i class="bi bi-cart"></i>
+                </div>
+              </div>
             </div>
-            <div class="text-field">
-              <input type="password" required class="form-input" />
-              <span></span>
-              <label>Password</label>
+          </section>
+          <section class="component">
+            <div class="card__container">
```

Comment:
you have used section for every component. it would be better If you have done the it using article or normal div.

If you dont want to change now please leave it. But please go through these articles to have better understanding of where to use section and where to use article

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/section

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/article

Reply:
made the changes. Will use the correct semantics from now on. Thanks a lot for sharing these resources as well.